tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,403 @@
+    assert_script_run("wget https://github.com/kubevirt/kubevirt/releases/download/v${kubevirt_ver}/local-block-storage.yaml");
+    assert_script_run("sed -i 's/node01/$hostname/g' local-block-storage.yaml");
+    assert_script_run("kubectl apply -f local-block-storage.yaml");
+
+    assert_script_run("kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/master/deploy/local-path-storage.yaml");
+    assert_script_run("kubectl -n local-path-storage wait deployment/local-path-provisioner --for condition=available --timeout=15m", timeout => 900);
+
+    # Check if the local block device is available
+    script_retry('losetup /dev/loop0', retry => 5, delay => 20, timeout => 120);
+
+    record_info('List host images', script_output('ls /tmp/hostImages/ -l -R'));
+    record_info('List local storage', script_output('ls /mnt/local-storage -R -l'));
+    record_info('Check the loop device "loop0"', script_output('losetup /dev/loop0'));
+    record_info('Check all loop devices', script_output('losetup -l -a'));
+
+    record_info('Set local storage class', '');
+    my $test_suite_config = '/usr/share/kube-virt/manifests/testing/default-config.json';
+    assert_script_run("sed -i '/storageClassLocal/s/local/local-path/' $test_suite_config");
+}
+
+sub run_virt_tests {
+    zypper_call('in ant-junit') if (script_run('rpmquery ant-junit'));
+
+    # Ensure Config::Tiny module installed
+    assert_script_run('cpan install Config::Tiny <<<yes', timeout => 300) if (script_run('cpan -l <<<yes | grep Config::Tiny') == 1);
+
+    my $result_dir = '/tmp/artifacts';
+    record_info('Create artifacts path', $result_dir);
+    assert_script_run("mkdir -p $result_dir");
+
+    # Run virt-tests command for each go test files
+    my $test_conf     = 'kubevirt-tests.conf';
+    my $parser_script = 'config_parser.pl';
+    assert_script_run("curl " . data_url("virt_autotest/kubevirt_tests/$test_conf") . " -o $test_conf");
+    assert_script_run("curl " . data_url("virt_autotest/kubevirt_tests/$parser_script") . " -o $parser_script");
+
+    my $test_suite_config = '/usr/share/kube-virt/manifests/testing/default-config.json';
+    my ($go_test, $skip_test, $extra_opt, $params);
+    my ($artifacts, $junit_xml, $test_log, $test_cmd);
+    my $retry_times = get_var('FAILED_RETRY');
+
+    # Workaround for bsc#1199448
+    my $node_helper = 'node-helper.yaml';
+    assert_script_run("curl " . data_url("virt_autotest/kubevirt_tests/$node_helper") . " -o $node_helper");
+    assert_script_run("kubectl apply -f $node_helper");
+
+    my $ginkgo_focus = get_var('GINKGO_FOCUS');
+    if ($ginkgo_focus) {
+        my $extra_opt = get_var('EXTRA_OPT');
+        $artifacts = "$result_dir/specific_test";
+        $junit_xml = "$result_dir/specific_test.xml";
+        $test_log  = "$result_dir/specific_test.log";
+        $test_cmd  = "virt-tests -ginkgo.regexScansFilePath=true ".
+                     "-ginkgo.focus='$ginkgo_focus' ".
```

Comment:
Do I understand correctly that when GINKOGO_FOCUS setting is given when triggering job, this will be all run in one virt-tests call?

Reply:
When setting GINKGO_FOCUS, virt-tests will call and run specific test cases, not all test suites in kubevirt-test.conf. I need to trigger a full test run after PR#15048 merged.

Reply:
I mean the tests filtered out by GINKOGO_FOCUS would  be run in a single virt-tests command, right? eg if pattern1|pattern2|pattern3 is given to GINKOGO_FOCUS , all 3 patterns tests would be run together? 
In addition to full test, I suggest to also test one more scenario of with GKINKGO_FOCUS set for multiple keywords.

Reply:
Yes, correct. Will trigger a verification run with the param GINKGO_FOCUS separately.